### PR TITLE
libnetworkit, networkit 8.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/networkit.rb [hunk 1]
```diff
@@ -4,10 +4,9 @@ class Networkit < Formula
   desc "Performance toolkit for large-scale network analysis"
   homepage "https://networkit.github.io"
   url "https://github.com/networkit/networkit.git",
-      tag:      "7.1",
-      revision: "4c6dcc4367b51005a34221242048609c357ffbd6"
+      tag:      "8.0",
+      revision: "bdbb926e7d3102c32a3fc54d4cfa769bfac822f4"
```
Comment (mitchblank) [inline Formula/networkit.rb:8]: Is there really any reason to keep pulling from git instead of using the tarball releases hosted on github?

Comment (chenrui333) [reply Formula/networkit.rb:8]: I don't have context for why it is using the tab and revision. Maybe the artifact issue? 🤷‍♂

Comment (SMillerDev) [reply Formula/networkit.rb:8]: Can you try switching?

Comment (chenrui333) [reply Formula/networkit.rb:8]: test it out in my local right now.

Comment (chenrui333) [reply Formula/networkit.rb:8]: this is the reason why it was using `git checkout`.
```
CMake Error at CMakeLists.txt:210 (message):
  Missing TLX library in extlibs/tlx Please run `git submodule update --init`
  to fetch the submodule.
```

Comment (chenrui333) [reply Formula/networkit.rb:8]: we do have tlx formula, so it should be good.

Comment (chenrui333) [reply Formula/networkit.rb]: works for me and tested out. 🎉

**Comment by chenrui333** [comment]
still the same error.

```
  BuildError: Failed executing: /usr/local/opt/python@3.9/bin/python3 -c import\ networkit\ as\ nk
G\ =\ nk.graph.Graph(3)
G.addEdge(0,1)
G.addEdge(1,2)
G.addEdge(2,0)
assert\ G.degree(0)\ ==\ 2
assert\ G.degree(1)\ ==\ 2
assert\ G.degree(2)\ ==\ 2
```

**Comment by chenrui333** [comment]
Since networkit bottling issue is already tracked [in here](https://github.com/Homebrew/homebrew-core/issues/65000), I think we can merge this PR as it is.

**Comment by fabratu** [comment]
Thanks @chenrui333 @SMillerDev for the help updating the package. As the maintainer of the first version of the package I can confirm the reason behind using github repo instead of the archive. Also confirmed: the suggested change works with the underlying build-system. There are plans to change the detection of `tlx` in a future update of `networkit` - in which case I will also post an update here.

As for 11.0. I tested the PR locally with 11.1 on Intel Kabylake. No problems there:

Test command:
```
❯ brew test-bot --only-formulae networkit
==> Using Homebrew/homebrew-test-bot 138ab86 (Merge pull request #548 from Homebrew/dependabot/bundler/rspec-mocks-3.10.1)
==> Testing Homebrew/brew 2.7.1-119-g4a90a5a5d (Merge pull request #10199 from reitermarkus/check-https-content)
==> Using Homebrew/homebrew-core 39af1529f3 (awsweeper: update 0.11.1 bottle.)

==> Running Formulae#detect_formulae!
    url             (undefined)
    origin/master   (undefined)
    HEAD            (undefined)
    diff_start_sha1 39af1529f356db9c9b20409e24ed763d8950f960
    diff_end_sha1   39af1529f356db9c9b20409e24ed763d8950f960
==> Testing Formula changes:
    added    (empty)
    modified (empty)
    deleted  (empty)

==> Running Formulae#formula!(networkit)
==> Determining dependencies...
==> brew link libnetworkit
==> brew fetch --retry swig
==> Determining dependents...
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
==> Starting build of networkit
==> brew fetch --retry networkit --build-bottle
==> brew uninstall --force networkit
==> brew install --only-dependencies --verbose --build-bottle networkit
==> brew install --verbose --build-bottle networkit
==> brew audit networkit --online --git --skip-style
==> brew bottle --verbose --json networkit
==> brew bottle --merge --write --no-commit ./networkit--8.0.big_sur.bottle.json
==> brew uninstall --force networkit
==> brew uninstall --force swig
==> brew install --only-dependencies ./networkit--8.0.big_sur.bottle.tar.gz
==> brew install ./networkit--8.0.big_sur.bottle.tar.gz
==> brew linkage --test networkit
==> brew install --only-dependencies --include-test networkit
==> brew test --verbose networkit
All steps passed!
```

OS-version:
```
system_profiler SPSoftwareDataType
Software:

    System Software Overview:
      System Version: macOS 11.1 (20C69)
    ...
```

CPU:
```
sysctl -n machdep.cpu.brand_string
Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
```

**Comment by chenrui333** [comment]
@fabratu 👋 thanks for testing out the formula on 11.0.

> As the maintainer of the first version of the package I can confirm the reason behind using github repo instead of the archive.

Can you elaborate a bit on why using tag/revision vs tarball file?

**Comment by chenrui333** [comment]
rebased to latest master and have another run.

**Comment by SeekingMeaning** [comment]
https://github.com/networkit/networkit/pull/666 😈

**Review by SeekingMeaning — APPROVED** [review]

**Pull request closed without merging**